porch/func/internal/podevaluator.go

```diff
@@ -65,30 +74,109 @@ func NewPodEvaluator(namespace, wrapperServerImage string) (Evaluator, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to create client: %w", err)
 	}
-	return &podEvaluator{
+
+	pe := &podEvaluator{
 		kubeClient:         cl,
 		namespace:          namespace,
 		wrapperServerImage: wrapperServerImage,
-	}, nil
+		gcScanInternal:     time.Duration(interval) * time.Second,
+		podTTL:             time.Duration(ttl) * time.Second,
+		imageEntrypoint:    map[string][]string{},
+		grpcConnectionPool: map[string]*grpc.ClientConn{},
```

Comment:
is the key for both maps the same?
If so, we could combine them into a single map, keyed by image and it contains the cached entrypoint info and the gRPC client to then call the function.